### ISSUE TITLE
Issues with LDAP3 Abstraction Layer while writing boolean values.

### ISSUE THREAD

**Issue by Sjd-Risca** [cannatag/ldap3#275]
I'm currently using in my software the version 2.1.1 of ldap3.

In this software I'm interfacing with the ldap server through ldap3 and its pythonic Abstraction Layer. So far so good (it made the editing so fast), until I introduced an object with a boolean attribute. In fact I get the following bug on the logic of the Abstraction Layer:
 - when reading the value is return as boolean (`<type 'bool'>`) like the following:

```
DN: o=test,dc=example,dc=tld - STATUS: Writable, Pending changes - READ TIME: 2016-12-27T23:42:20.660217
    objectClass: testClass
                 top
    myBoolAttribute: True
```

 - when writing if I try to do a change inserting a pythonic boolean value (like the following) I get an _LDAPInvalidAttributeSyntaxResult_:

```
DN: o=test,dc=example,dc=tld - STATUS: Writable, Pending changes - READ TIME: 2016-12-27T23:42:20.660217
    objectClass: testClass
                 top
    myBoolAttribute: True
              CHANGES: [('MODIFY_REPLACE', [False])]
```
 * instead it works if I use, in setting the boolean value, a string like 'FALSE' or 'TRUE' (exactly as I would expect to do for ldap syntax), like:

```
              CHANGES: [('MODIFY_REPLACE', ['FALSE'])]
```

I think that it's really cool that reading the value into the Abstraction Layer I get back, for boolean type [1], a boolean value. But for consistency it should then let me also write on the attribute using boolean values. I would rather prefer a consistent behavior, let me use both for reading and for writing only string or only boolean.

[1] https://tools.ietf.org/html/rfc4517#section-3.3.3

**Comment by cannatag** [comment]
Hi, I'm aware of this inconsistency. The Abstraction Layer has a "validator" feature, for each type of data sent, but it's still not functional for all data types. Will work on this in the upcoming release of ldap3.

**Comment by cannatag** [comment]
Can you check code in dev? It should work for boolean values now. It should accept True, False and their string representation.

**Comment by cannatag** [comment]
Will be released in 2.2

**Comment by cannatag** [comment]
included in v2.2.0

**Comment by Sjd-Risca** [comment]
Thank you,
tested against v2.2.0 and works like a charm.
